README.md

```diff
@@ -479,6 +479,48 @@ Contributions of additional uses cases on top of ECS are welcome.
 * *Use prefixes.* Fields must be prefixed except for the base fields. For example all `host` fields are prefixed with `host.`. See `dot` notation in FAQ for more details.
 * Do not use abbreviations. (A few exceptions like `ip` exist.)
 
+## Understanding ECS conventions
+
+### Multi-fields text indexing
+
+ElasticSearch can index text multiple ways:
+
+* `text` indexing allows for full text search, or searching arbitrary words that
+  are part of the field.
+* `keyword` indexing allows for exact match search (much faster) and allows for
+  aggregations (what Kibana visualizations are built on).
+
+In some cases only one type of indexing makes sense for a field. E.g. no need to
+do full text search on an id, and nobody needs to do an exact match search on
+a 2kb stack trace.
+
+However there are cases where both types of indexing can be useful.
+As an example, log messages can sometimes be short enough that it makes sense
+to sort them by frequency (that's an aggregation). They can also be long and
+varied enough that full text search can be useful on them.
+
+Whenever both types of indexing are helpful, we use multi-fields indexing. The
+convention used is the following:
+
+* `foo`: `text` indexing. The top level of the field (its plain name) is used
```

Comment:
Will this convention change any of the existing fields we have? Mainly curious.

In general +1 on the convention.

Reply:
Well most of the non-multi-fields text fields are actually `keyword`. That's why I follow all this immediately with a mention to that effect.

But any time we do multi-field, we need to follow this convention, otherwise it will get confusing :-)

Also, this PR is a side-effect of the thinking/discussion that's happening around https://github.com/elastic/ecs/issues/87. And obviously we will need to tweak the wording here if we go with `.keyword` instead of `.raw`, but I'm fine with that :-)